recipe/meta.yaml

```diff
@@ -11,19 +11,21 @@ source:
 
 
 build:
-  number: 0
+  number: 1
 
 requirements:
   build:
     - python                                 # [build_platform != target_platform]
     - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+    - libprotobuf                            # [build_platform != target_platform]
     - {{ compiler('rust') }}
+    - {{ stdlib('c') }}
     - cargo-bundle-licenses
-    - maturin >=0.15,<0.16
-    - libprotobuf =3
+    - maturin
   host:
+    - libprotobuf
     - python
-    - maturin >=0.15,<0.16
+    - maturin
```

Comment:
Generally in favor of unpinning the maturin dependency as this is quite old but is there a particular reason we need to bump?

Reply:
The reason is that there are no python 3.12 builds for maturin 0.15 (or one of its dependencies, I didn't dig deeper). 🙃

Reply:
Opened https://github.com/apache/datafusion-python/issues/701 to discuss this with the maintainers, which led to a bumped maturin constaint:

- https://github.com/apache/datafusion-python/pull/725

With that in, would it make sense to update this to match that new constraint?

```
    - maturin >=1.5.1,<1.6.0
```

Reply:
1.6 is out already, so you should be easily able to test it. If you (or the upstream developers) insist on pinning something _this_ tightly, you also assume responsibility to keep it reasonably up to date. 🙃 

In any case, I added the cap you suggested.

Reply:
Yeah in general I'm not too involved with the ADP project itself so didn't really have much control over this pinning 😕 but hoping that we can maybe grab a few active maintainers there to be maintainers on this feedstock so that these considerations are taken with more gravity at least